Shared/Samples/Add custom dynamic entity data source/AddCustomDynamicEntityDataSourceView.swift

```diff
@@ -0,0 +1,164 @@
+        )
+        
+        info.spatialReference = .wgs84
+        
+        let customDataSource = CustomDynamicEntityDataSource(info: info) { VesselFeed() }
+        
+        _dynamicEntityLayer = .init(initialValue: DynamicEntityLayer(dataSource: customDataSource))
+        
+        let trackDisplayProperties = dynamicEntityLayer.trackDisplayProperties
+        trackDisplayProperties.showsPreviousObservations = true
+        trackDisplayProperties.showsTrackLine = true
+        trackDisplayProperties.maximumObservations = 20
+        
+        let labelDefinition = LabelDefinition(
+            labelExpression: SimpleLabelExpression(simpleExpression: "[VesselName]"),
+            textSymbol: TextSymbol(color: .red, size: 12)
+        )
+        labelDefinition.placement = .pointAboveCenter
+        
+        dynamicEntityLayer.addLabelDefinition(labelDefinition)
+        dynamicEntityLayer.labelsAreEnabled = true
+        
+        map.addOperationalLayer(dynamicEntityLayer)
+    }
+    
+    var body: some View {
+        MapViewReader { proxy in
+            MapView(map: map)
+                .onSingleTapGesture { screenPoint, _ in
+                    tappedScreenPoint = screenPoint
+                }
+                .callout(placement: $calloutPlacement.animation(.default.speed(2))) { placement in
+                    let attributes = (placement.geoElement as! DynamicEntityObservation).attributes
+                    VStack(alignment: .leading) {
+                        // Display all the attributes in the callout.
+                        ForEach(attributes.sorted(by: { $0.key < $1.key }), id: \.key) { item in
+                            Text("\(item.key): \(String(describing: item.value))")
+                        }
+                    }
+                }
+                .task(id: tappedScreenPoint) {
+                    guard let tappedScreenPoint,
+                          let identifyResult = try? await proxy.identify(
+                            on: dynamicEntityLayer,
+                            screenPoint: tappedScreenPoint,
+                            tolerance: 2
+                          ) else {
+                        // Hides the callout.
+                        calloutPlacement = nil
+                        
+                        return
+                    }
+                    
+                    // Set the callout placement to the observation that was tapped on.
+                    calloutPlacement = identifyResult.geoElements.first.map { .geoElement($0) }
+                }
+        }
+    }
+}
+
+/// The vessel feed that is emitting custom dynamic entity events.
+private struct VesselFeed: CustomDynamicEntityFeed {
+    let events: AsyncThrowingStream<CustomDynamicEntityFeedEvent, Error> = .init { continuation in
+        Task.detached {
+            do {
+                let fileHandle = try FileHandle(forReadingFrom: .selectedVesselsDataSource)
+                let decoder = JSONDecoder()
```

Comment:
FYI: Apple has said before the coders are not thread safe and therefore shouldn't be re-used in asynchronous contexts.